python/lsst/afw/table/baseRecord.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import
+
+from ._baseRecord import *
+
+def _baseRecordSet(self, key, value):
+    """
+    Given a string or `lsst.afw.table.Key`, find the `BaseRecord` for the given key and set its value.
+    """
+    if isinstance(key, basestring):
```

Comment:
`basestring` is not Python 3 compatible. However, the `future` module provides a version, so if you import that the code will be compatible with Python 2.7 and Python 3